plugins/lime-plugin-locate/src/communityLayer.js

```diff
@@ -0,0 +1,75 @@
+const isObject = (obj) => typeof obj === 'object' && obj !== null;
+
+const sameLocation = (coord1, coord2) => Number(coord1.lon) === Number(coord2.lon) && Number(coord1.lat) === Number(coord2.lat);
+
+const removeDuplicates = (collection, getIdentifier) => {
+    const identifierState = {};
+    return collection.filter((value) => {
+        const identifier = String(getIdentifier(value));
+        if (identifierState[identifier]) {
+            return false;
+        }
+        identifierState[identifier] = true;
+        return true;
+    });
+};
+
+const coordsToPoint = ({ coordinates, hostname }) => ({ type: 'Feature', geometry: { type: 'Point', coordinates: [Number(coordinates.lon), Number(coordinates.lat)] }, properties: { name: hostname } });
+
+const coordspairToLineString = (coordinates) => ({ type: 'Feature', geometry: { type: 'LineString', coordinates } });
+
+export function getCommunityLayer(hostname, stationLocation, nodeshash) {
+    //Run only if leaflet is loaded
+    let nodes = Object.values(nodeshash)
+        .filter(node => isObject(node))
+        .filter(n => !isNaN(Number(n.coordinates.lat))); // Filter nodes without coordinates
+    const thisnode = nodes.filter(n => n.hostname === hostname)
+    if (thisnode) {
+        thisnode[0].coordinates = stationLocation
+    }
+    // geomac being the hash of locations of nodes indexed by mac
+    let geomac = nodes     // to those nodes
+        .filter(n => n.macs) // that actually have a macs list
```

Comment:
Is there any case were a node has no MACs?
I think not, as in order to be part of the map it needs to be connected to others, so it will always have at least one interface.